Packs/SilentPush/Integrations/SilentPush/SilentPush.py

```diff
@@ -104,20 +103,290 @@ def _http_request(self, method: str, url_suffix: str, params: dict = None, data:
+                    query: Optional[str] = None, 
+                    start_date: Optional[str] = None,
+                    end_date: Optional[str] = None,
+                    risk_score_min: Optional[int] = None,
+                    risk_score_max: Optional[int] = None,
+                    limit: int = 100) -> dict:
+        """
+        Search for domains with optional filters.
+        
+        Args:
+            query (str, optional): Search query string (e.g., domain pattern, keywords)
+            start_date (str, optional): Start date for domain registration (ISO8601 format)
+            end_date (str, optional): End date for domain registration (ISO8601 format)
+            risk_score_min (int, optional): Minimum risk score filter
+            risk_score_max (int, optional): Maximum risk score filter
+            limit (int, optional): Maximum number of results to return (default: 100)
+            
+        Returns:
+            dict: A dictionary containing the search results
+        """
+        demisto.debug(f'Searching domains with query: {query}')
+        url_suffix = 'explore/domain/search'
+        
+        params = {k: v for k, v in {
+            'domain': query,
+            'start_date': start_date,
+            'end_date': end_date,
+            'risk_score_min': risk_score_min,
+            'risk_score_max': risk_score_max,
+            'limit': limit
+        }.items() if v is not None}
+        
+        try:
+            return self._http_request('GET', url_suffix, params=params)
+        except Exception as e:
+                demisto.error(f"Error in search_domains API request: {str(e)}")
+                
+    def list_domain_infratags(self, domains: list, cluster: bool = False, mode: str = 'live', match: str = 'self', as_of: Optional[str] = None) -> dict:
+        """
+        Get infratags for multiple domains with optional clustering and additional filtering options.
+
+        Args:
+            domains (list): A list of domains to retrieve infratags for.
+            cluster (bool, optional): Whether to cluster the results. Defaults to False.
+            mode (str, optional): Mode for the lookup, either 'live' (default) or 'padns'.
+            match (str, optional): Handling of self-hosted infrastructure, either 'self' (default) or 'full'.
+            as_of (str, optional): Date or timestamp for filtering the data.
+
+        Returns:
+            dict: A dictionary containing infratags for the provided domains.
+        """
+        demisto.debug(f'Fetching infratags for domains: {domains} with cluster={cluster}, mode={mode}, match={match}, as_of={as_of}')
+        
+        
+        url = 'explore/bulk/domain/infratags'  
+        
+      
+        payload = {
+            'domains': domains
+        }
+        params = {
+            'mode': mode,
+            'match': match,
+            'clusters': 1 if cluster else 0  
+        }
+        
+        if as_of:
+            params['as_of'] = as_of
```

Comment:
@karand-metron 
```suggestion
            params['as_of'] = as_of
```This includes epoch/date/sec, please add check for it.